Casks/circleci-runner.rb

```diff
@@ -29,6 +29,14 @@
   service "#{Dir.home}/Library/LaunchAgents/com.circleci.runner.plist"
 
   preflight do
+    # Create the necessary directories prior to installation
+    dirs = [configDir, launchAgentDir, logDir, workingDir]
+    dirs.each do |dir|
+      if not File.exist?(dir)
+        system_command 'mkdir', args: ['-p', dir]
+      end
+    end
```

Comment:
I'm not sure if matters, but we could move some of these dirs back down to `postflight`.